core/src/traits/arithmetic_types_traits.hpp

```diff
@@ -0,0 +1,49 @@
+/*
```

Comment:
I think `arithmetic.hpp` is a proper name since the file is contained in a `traits` folder.